src/VisualStudio/Xaml/Impl/Implementation/LanguageServer/Handler/XamlSolutionProvider.cs

```diff
@@ -27,17 +25,14 @@ public XamlSolutionProvider(ILspSolutionProvider lspSolutionProvider, XamlProjec
             _projectService = projectService;
         }
 
-        public Solution GetCurrentSolutionForMainWorkspace()
-        {
-            return _solutionProvider.GetCurrentSolutionForMainWorkspace();
-        }
-
-        public ImmutableArray<Document> GetDocuments(Uri documentUri)
+        public (DocumentId?, Solution) FindDocumentAndSolution(TextDocumentIdentifier? textDocument, string? clientName)
```

Comment:
maybe we can do this in a followup, this seems like we could just move this to an extension method or something and then remove ILspSolutionProvider

Reply:
I thought about this, but it felt odd having an extension method on a workspace registration service, that returned documents. I could be convinced otherwise, but I'd rather follow up and just remove the solution provider altogether by having this method only need to be called from inside the request queue.